packages/client/src/getBalances.ts

```diff
@@ -217,11 +223,16 @@ export const getBalances = async function (
           !["btc_testnet", "btc_mainnet"].includes(token.chain_name)
       )
       .map(async (token) => {
-        const rpc = await this.getEndpoint("evm", token.chain_name);
-        const provider = new ethers.providers.StaticJsonRpcProvider(rpc);
-        const balance = await provider.getBalance(evmAddress);
-        const formattedBalance = formatUnits(balance, token.decimals);
-        balances.push({ ...token, balance: formattedBalance });
+        const chainLabel = Object.keys(this.getChains()).find(
+          (key) => this.getChains()[key].chain_id === parseInt(token.chain_id)
+        );
```

Comment:
nit: Object.values may make this a bit simpler to read without needing to call this.getChains() twice

Reply:
If you mean something like this,

```ts
        const chain = Object.values(this.getChains()).find(
          (chain) => chain.chain_id === parseInt(token.chain_id)
        );
```

Doesn't work, because I need to get the key, not the value.

Reply:
I see, alternatively Object.entries would allow to avoid calling getChains twice while still having the key at hand, but I haven't checked what getChains does inside so it's probably a very minor nit